packages/es-dev-server/src/middleware/compatibility.js

```diff
@@ -94,77 +94,98 @@ export function createCompatibilityMiddleware(cfg) {
      * Modules are loaded through es-module-shims for modern browsers, for browsers which
      * don't support modules the code is loaded by systemjs and compiled to es5.
      */
-    if (ctx.url === cfg.appIndex) {
-      const lastModified = ctx.response.headers['last-modified'];
-
-      // return cached index.html if it did not change
-      if (cachedIndexHTML) {
-        if (cachedIndexHTML.lastModified === lastModified) {
-          ctx.body = cachedIndexHTML.body;
-          return;
-        }
-      }
 
-      // read served index.html
-      const indexHTMLString = await getBodyAsString(ctx.body);
+    //  only look for html files
+    if (!ctx.url.endsWith('/') && !ctx.url.endsWith('.html')) {
+      return;
+    }
 
-      // extract input files from index.html
-      const resources = extractResources(indexHTMLString);
-      if (resources.inlineModules.length > 0) {
-        throw new Error(
-          `Compatibility cannot handle "inline" modules (modules without a src attribute). Place your js code in a separate file.`,
-        );
-      }
+    if (ctx.status < 200 || ctx.status >= 300) {
+      return;
+    }
 
-      if (!resources.jsModules || resources.jsModules.length === 0) {
-        throw new Error(
-          `Compatibility mode requires at least one <script type="module" src="..."> in your index.html.`,
-        );
-      }
+    // if url is the app index we know for sure we need to inject polyfills. otherwise, check if
+    // this is a html document we are serving.
+    let indexHTMLString;
+    if (ctx.url !== cfg.appIndex) {
+      indexHTMLString = await getBodyAsString(ctx.body);
 
-      const files = resources.jsModules.map(e => e.replace('./', ''));
-
-      // create a new index.html with injected polyfills and loader script
-      const createResult = createIndexHTML(resources.indexHTML, {
-        entries: {
-          type: 'module',
-          files,
-        },
-        legacyEntries: modernOnly
-          ? undefined
-          : {
-              type: 'system',
-              files,
-            },
-        polyfills: modernOnly ? modernPolyfills : legacyPolyfills,
-        minify: false,
-        preload: false,
-      });
-
-      let { indexHTML } = createResult;
-
-      // if there were any importmaps, they were extracted. re-add them as an importmap shim
-      indexHTML = injectImportMaps(indexHTML, resources, 'importmap-shim');
-
-      if (!modernOnly) {
-        // if we need to support legacy browsers, also add systemjs-importmap
-        indexHTML = injectImportMaps(indexHTML, resources, 'systemjs-importmap');
-
-        // inject systemjs resolver which appends a query param to trigger es5 compilation
-        indexHTML = indexHTML.replace('</body>', `${systemJsLegacyResolveScript}</body>`);
+      if (!indexHTMLString.includes('<html') && !indexHTMLString.includes('</html>')) {
```

Comment:
Want to double check the intersection of this test with this issue: https://github.com/open-wc/open-wc/issues/526

Reply:
This code doesn't touch the webpack plugin. Here we just need to check if it's an `index.html` file being served in order to determine whether we can inject polyfills in there.